core/src/test/java/kafka/test/annotation/ClusterTest.java

```diff
@@ -33,7 +33,7 @@
 @Retention(RUNTIME)
 @TestTemplate
 public @interface ClusterTest {
-    Type clusterType() default Type.DEFAULT;
+    Type[] clusterTypes() default {};
```

Comment:
this bit is the one I'm still going around. I totally like the direction of the PR, removing the type ALL and DEFAULT , but then it brings the need to specify the full list of cluster types on every test that needs them all, so wonder if having the full list as default here would be a good complementary change? It would mean that we have well defined `ClusterTypes` as introduced by this PR, but every test running on `ClusterTest` would run for all types, unless specified. That removes lots of `clusterTypes = {Type.ZK, Type.KRAFT, Type.CO_KRAFT}`, it's probably the default behaviour we want for a test so that no ones accidentally forgets to run on any specific cluster type, and makes it easier to maintain (keeping the full list only in this single place, not on every test). Thoughts?

Reply:
> so wonder if having the full list as default here would be a good complementary change?
> It would mean that we have well defined ClusterTypes as introduced by this PR, but every test running on ClusterTest would run for all types, unless specified. That removes lots of clusterTypes = {Type.ZK, Type.KRAFT, Type.CO_KRAFT}, it's probably the default behaviour we want for a test so that no ones accidentally forgets to run on any specific cluster type, and makes it easier to maintain (keeping the full list only in this single place, not on every test). Thoughts?

I love this idea!

Reply:
oh, I assumed we are talking about default value of `ClusterTestDefaults` :)

Reply:
`ClusterTestDefaults` makes sense to me. Honestly I wasn't thinking of it when suggesting having a default with all the types, simply because I didn't remember we had it. My point was more about having a default with all types, agree with you that the right place for that default should be ClusterTestDefaults